astropy/coordinates/distances.py

```diff
@@ -162,6 +162,7 @@ def __new__(
                     unit = u.pc
 
         elif parallax is not None:
+            parallax = u.Quantity(parallax, copy=False)
```

Comment:
Wouldn't this improve performance?
```suggestion
            parallax = u.Quantity(parallax, copy=False, subok=True)
```